### ISSUE TITLE
Not working for nested forms, missing index

### ISSUE THREAD

**Issue by jokklan** [refile/refile#335]
Hi

I can't remove or update existing images on a nested model after first upload. The problem seems to be that there is missing a index on the html tag name attribute on the hidden field.

Example:
```erb
<%= form_for @user do |f| %>
  <%=  f.fields_for :images do |image_form| %>
    <%= image_form.attachment_field :file %>
  <% end %>
<% end %>
```
results in something like:
```html
<form id="edit_user_1" enctype="multipart/form-data" action="/users/1" method="post">
  <!-- This is where the index [1] is missing in the name attribute -->
  <input value="null" type="hidden" name="user[image_attributes][file]"> 

  <input name="user[image_attributes][1][file]" id="user_image_attributes_1_file">
</form>
```

I'm not sure how this can be solved, i looked https://github.com/refile/refile/blob/v0.5.5/lib/refile/rails/attachment_helper.rb but can't figure out why `hidden_field` and `file_field` behaves differently.

UPDATE:
To clarify, this problems is only for nested attributes through a `have_many` association.
```ruby
class User < ActiveRecord::Base
  has_many :images

  accepts_nested_attributes_for :images
end
```

**Comment by simonwh** [comment]
I have this exact same problem. Any ideas on what we can do to fix it?

**Comment by simonwh** [comment]
@jnicklas Do you have any ideas on how to go about solving this?

**Comment by sidtri** [comment]
It seems fine for me to adding, updating records. Can you give some more content related to this issue.

```html
<div class="field">
    
     <input type="text" value="df" name="user[post_attributes][name]" id="user_post_attributes_name">
     <input value="{}" data-reference="8c74cccf68793046d2072efa23faeb6a" type="hidden" name="user[post_attributes][image]"><input data-reference="8c74cccf68793046d2072efa23faeb6a" type="file" name="user[post_attributes][image]" id="user_post_attributes_image">
<input type="hidden" value="1" name="user[post_attributes][id]" id="user_post_attributes_id">  </div>
```

Values are populating fine for me

**Comment by jokklan** [comment]
@sidtri you only have óne related record (`has_one`/`belongs_to`), the problem is when you have many nested models.

**Comment by jnicklas** [comment]
Is this issue solved by #406? Can we close this?

**Comment by jokklan** [comment]
@jnicklas yes it is, thanks! I'll close it :)
